lib/ecto/schema.ex

```diff
@@ -453,7 +453,7 @@ defmodule Ecto.Schema do
   `@primary_key` attribute.
   """
   defmacro embedded_schema([do: block]) do
-    schema(nil, false, :binary_id, block)
+    schema(__ENV__, nil, false, :binary_id, block)
```

Comment:
This should be `__CALLER__`. I recommend you to check the line in your test as it would have caught this error. :)

Reply:
I cannot make `Code.compile_quoted` to return proper lines so I fall back to compiling string instead.